packages/edit-post/src/components/header/index.js

```diff
@@ -79,18 +79,17 @@ function Header( {
 export default compose(
 	withSelect( ( select ) => ( {
 		hasActiveMetaboxes: select( 'core/edit-post' ).hasMetaBoxes(),
-		hasBlockSelection: !! select( 'core/editor' ).getBlockSelectionStart(),
 		isEditorSidebarOpened: select( 'core/edit-post' ).isEditorSidebarOpened(),
 		isPublishSidebarOpened: select( 'core/edit-post' ).isPublishSidebarOpened(),
 		isSaving: select( 'core/edit-post' ).isSavingMetaBoxes(),
 	} ) ),
-	withDispatch( ( dispatch, { hasBlockSelection } ) => {
+	withDispatch( ( dispatch, ownProps, select ) => {
+		const { getBlockSelectionStart } = select( 'core/editor' );
 		const { openGeneralSidebar, closeGeneralSidebar } = dispatch( 'core/edit-post' );
-		const sidebarToOpen = hasBlockSelection ? 'edit-post/block' : 'edit-post/document';
+
 		return {
-			openGeneralSidebar: () => openGeneralSidebar( sidebarToOpen ),
+			openGeneralSidebar: () => openGeneralSidebar( getBlockSelectionStart() ? 'edit-post/block' : 'edit-post/document' ),
 			closeGeneralSidebar: closeGeneralSidebar,
-			hasBlockSelection: undefined,
```

Comment:
I don't really understand why we felt the need to assign this to `undefined` previously? In other words, why not just let the `hasBlockSelection` from `withSelect` pass through.

It's moments like these where I start to re-doubt making this `select` available, because it becomes less clear what to use when.

Reply:
Yes, overriding with `undefined` doesn't help here.

I still see value in using `select` here. You won't see re-renders of `Header` component triggered by `hasBlockSelection` value change.